include/mbedtls/config.h

```diff
@@ -3339,6 +3339,4 @@
 #include MBEDTLS_USER_CONFIG_FILE
 #endif
 
-#include "check_config.h"
```

Comment:
If an application is currently using `#include <mbedtls/config.h>`, not including `<limits.h>` explicitly, and relying on the inclusion of `limits.h`, this change will break it. Sure, it's a bug in the application, but I don't think exposing this bug is acceptable for a change to an LTS branch.

This change is not necessary to fix the bug that applications that _don't_ include `check_config.h` don't build due to the use of symbols from `limit.h` inside the library. So it shouldn't be in the backport.